src/napari_spatialdata/_model.py

```diff
@@ -85,10 +85,11 @@ def get_obs(
         if name not in self.adata.obs.columns:
             raise KeyError(f"Key `{name}` not found in `adata.obs`.")
         if name != self.instance_key:
-            adata_obs = self.adata.obs[[self.instance_key, name]]
+            adata_obs = self.adata.obs[[self.instance_key, name]].copy()
```

Comment:
is this copy required because of the otherwise automated conversion to categorical when plotting?

Reply:
Good point, I could remove it. It was because I was setting the index in-place.